site/content/blog/eth2-dev-chain.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Setting up private Ethereum 2.0 development chain
+date: 2020-03-26T09:00:00-00:00
+image: /images/blog/ethereum20-validator.png
+draft: false
+keywords:
+- ethereum 2.0
+- proof of staking
+- dev
+- lighthouse
+- chainguardian
+description: Learn how to setup eth2 development chain using lighthouse client. 
+---
+
+> Disclaimer: This article is highly technical intended for developers and eth2 enthusiasts.
+
+### Why would I setup private eth2 chain?
+
+If you had any experience with development on eth1, you probably already know answer to this question so 
+feel free to skip ahead.
+
+At the moment, eth2 has a bunch short lived single client testnets and [one long lived](https://prylabs.net/) with
+real stable client testnet still TBD. So, yeah, you can use [Prysmatic's Sapphire Testnet](https://prylabs.net/) but
+beware it will take some time to sync and if you wanna test some validator related operations, you will have to wait
+couple of hours to become active validator. Imagine, waiting couple of hours just to find out your app has bug 
+with validator related operation and now you have to do everything all over.
+
+Our motivation for this setup was [ChainGuardian](https://github.com/nodefactoryio/chainguardian), a desktop validator
+client. Since we need to test whole onboarding process together with validation itself, dev chain is a must have.
+
+### Choosing client
+
+As you might already know, there is [quite a few teams](https://docs.ethhub.io/ethereum-roadmap/ethereum-2.0/eth2.0-teams/teams-building-eth2.0/)
+developing eth2 clients each focusing on specific features (security, enterprise, resource constrained devices, browser).
+
+We went with [Lighthouse client](https://github.com/sigp/lighthouse) as their API is closely matched to 
+[api standardization efforts](https://github.com/ethereum/eth2.0-apis) and other clients. Also, they seemed most
+stable at the moment.
+
+### Step 1: Setup private Eth1 chain
+
+We will use docker and docker-compose throughout this article to orchestrate various pieces of architecture.
+
+First, we need eth1 account that will serve purpose of signer in POA chain and it will be used to deploy deposit contract
+and validator deposits. Store private key of that account in `signer.key` file.
+
+We will also need:
+
+- genesis.json - genesis file containing params of our POA chain
+- passwd.txt - some file containing some dummy password for geth to encrypt our account
+
+You can read more about setting up POA chain and generating genesis file [here](https://hackernoon.com/setup-your-own-private-proof-of-authority-ethereum-network-with-geth-9a0a3750cda8)
```

Comment:
I think that linking `here` is not considered good SEO practice. so maybe this would be:

`If you want to learn more about custom POA chain and generating genesis file, you can go through tutorial on how to [setup your own private POA network with geth](https://hackernoon.com/setup-your-own-private-proof-of-authority-ethereum-network-with-geth-9a0a3750cda8)`